bindings/rust/s2n-tls/src/config.rs

```diff
@@ -585,6 +592,15 @@ impl Builder {
             }
         }
 
+        // If OCSP has not been explicitly requested, turn off OCSP. This is to prevent the `trust_location()` function
+        // from automatically enabling `OCSP` due to the legacy behavior of `s2n_config_set_verification_ca_location`
+        if !self.enable_ocsp {
+            unsafe {
+                s2n_config_set_status_request_type(self.as_mut_ptr(), s2n_status_request_type::NONE)
+                    .into_result()?
```

Comment:
Since this override is specific to the weird behavior of `trust_location()`/`s2n_config_set_verification_ca_location()`, would it make sense for this to go in `trust_location()` rather than `build()`? Or is there a reason to put it in `build()`?

Reply:
Good call. I was thinking there might be some issue depending on the order in which one called `enable_ocsp()` and `'trust_location()`, but thinking through it there shouldn't be a problem and it makes more sense to limit this legacy behavior to the method that causes it.

If someone calls `enable_ocsp()` before calling `trust_location()`, the first function will enable OSCP and the second will leave the OCSP status enabled. If `trust_location()` is called first, OSCP status will be disabled after the first function, but enabled by the second function.